client/packages/programs/src/JsonForms/components/DateOfBirth.tsx

```diff
@@ -101,21 +100,25 @@ const UIComponent = (props: ControlProps) => {
               },
             }}
           />
-          <Box
-            flex={0}
-            style={{ textAlign: 'end' }}
-            flexBasis={FORM_LABEL_COLUMN_WIDTH}
-          >
+
+          <Box flex={0} style={{ textAlign: 'end' }}>
             <FormLabel sx={{ fontWeight: 'bold' }}>{t('label.age')}:</FormLabel>
           </Box>
-          <Box flex={0}>
-            <NumericTextInput
-              value={age}
-              sx={{ width: 65 }}
-              onChange={onChangeAge}
-              disabled={!props.enabled}
-            />
-          </Box>
+          {(age ?? 1 >= 1) ? (
+            <Box flex={0}>
+              <NumericTextInput
+                value={age}
+                sx={{ width: 65 }}
+                onChange={onChangeAge}
+                disabled={!props.enabled}
+              />
+            </Box>
+          ) : (
+            <Typography fontSize="85%" whiteSpace="nowrap">
+              {months > 0 && t('label.age-months', { count: months })}{' '}
```

Comment:
I would recommend putting the separator (comma and space) into the 'month' translated string then you don't need the `{' '}` and it is truly conditional and more flexible.

Reply:
Yup, which means we'll definitely need to keep the "days" part, even if it's 0 (re: Andrei's earlier comment)

Reply:
c482d1438